src/services/item/plugins/file/service.ts

```diff
@@ -72,67 +69,66 @@ class FileItemService {
     );
 
     if (currentStorage + size > this.options.maxMemberStorage) {
-      throw new StorageExceeded();
+      throw new StorageExceeded(currentStorage + size);
     }
   }
 
   async upload(
-    actor: Actor,
-    repositories: Repositories,
-    files: (UploadedFile & { description?: string })[],
-    parentId?: string,
+    actor,
+    repositories,
+    {
+      description,
+      parentId,
+      filename,
+      mimetype,
+      fields,
+      stream,
+    }: {
+      description?: string;
+      parentId?: string;
+      filename;
+      mimetype;
+      fields?: MultipartFields;
+      stream: Readable;
+    },
   ) {
-    if (!actor) {
-      throw new UnauthorizedMember(actor);
+    const filepath = this.buildFilePath(); // parentId, filename
+    // compute body data from file's fields
+    if (fields) {
+      const fileBody = Object.fromEntries(
+        Object.keys(fields).map((key) => [
+          key,
+          (fields[key] as unknown as { value: string })?.value,
+        ]),
+      );
     }
+    // check member storage limit
+    await this.checkRemainingStorage(actor, repositories);
 
-    // check rights
-    if (parentId) {
-      const item = await repositories.itemRepository.get(parentId);
-      await validatePermission(repositories, PermissionLevel.Write, actor, item);
+    // duplicate stream to use in thumbnails
+    const streamForThumbnails = new PassThrough();
+    if (MimeTypes.isImage(mimetype)) {
+      stream.pipe(streamForThumbnails);
     }
```

Comment:
Can a stream be used by 2 different consumers? From my understanding, the stream here is piped both to the file upload and to the thumbnail upload. Does this actually create a copy of the stream data? What if the consumers don't use the data at the same rate?